x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/ComputeService.java

```diff
@@ -513,6 +532,7 @@ private void lookupDataNodes(
         QueryBuilder filter,
         Set<String> concreteIndices,
         OriginalIndices originalIndices,
+        EsqlExecutionInfo executionInfo,
```

Comment:
Can we avoid mutating executionInfo and instead return a list of data nodes and shard information?

Reply:
Good idea. I have made a go of that in [this commit](https://github.com/elastic/elasticsearch/pull/112472/commits/29b996bf96249c9a12561f6ad36846fd08511cec). Let me know what you think of that approach instead.